File: staffapp/src/main/java/com/example/staffapp/ui/components/StaffComponents.kt
package com.example.staffapp.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.ColumnScope
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.KeyboardArrowRight
import androidx.compose.material.icons.filled.ErrorOutline
import androidx.compose.material.icons.filled.Inbox
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.FilledTonalButton
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.FilterChip
import androidx.compose.material3.FilterChipDefaults
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.example.staffapp.ui.theme.StaffAccentBlue
import com.example.staffapp.ui.theme.StaffError
import com.example.staffapp.ui.theme.StaffOnSurfaceVariant
import com.example.staffapp.ui.theme.StaffPrimary
import com.example.staffapp.ui.theme.StaffSuccess
import com.example.staffapp.ui.theme.StaffWarning
import com.example.staffapp.ui.work.ActionUi
import com.example.staffapp.ui.work.BadgeColor
import com.example.staffapp.ui.work.DayChipUi
import com.example.staffapp.ui.work.ListCardUi
import com.example.staffapp.ui.work.MetricUi

@Composable
fun StaffScreenContainer(
    contentPadding: PaddingValues = PaddingValues(16.dp),
    content: @Composable () -> Unit,
) {
    LazyColumn(
        modifier = Modifier.fillMaxSize(),
        contentPadding = contentPadding,
        verticalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        item { content() }
    }
}

@Composable
fun StaffSectionTitle(title: String, modifier: Modifier = Modifier) {
    Text(
        text = title,
        style = MaterialTheme.typography.titleMedium,
        fontWeight = FontWeight.Bold,
        modifier = modifier.padding(top = 4.dp, bottom = 8.dp),
    )
}

@Composable
fun StaffHeroCard(
    title: String,
    subtitle: String,
    modifier: Modifier = Modifier,
) {
    Card(
        modifier = modifier.fillMaxWidth(),
        shape = RoundedCornerShape(20.dp),
        colors = CardDefaults.cardColors(containerColor = StaffPrimary),
        elevation = CardDefaults.cardElevation(defaultElevation = 4.dp),
    ) {
        Column(modifier = Modifier.padding(20.dp)) {
            Text(
                text = title,
                style = MaterialTheme.typography.headlineSmall,
                fontWeight = FontWeight.Bold,
                color = Color.White,
            )
            Spacer(modifier = Modifier.height(6.dp))
            Text(
                text = subtitle,
                style = MaterialTheme.typography.bodyMedium,
                color = Color.White.copy(alpha = 0.9f),
            )
        }
    }
}

@Composable
fun StaffMetricsRow(metrics: List<MetricUi>, modifier: Modifier = Modifier) {
    if (metrics.isEmpty()) return
    LazyRow(
        modifier = modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.spacedBy(10.dp),
    ) {
        items(metrics) { metric ->
            Card(
                modifier = Modifier.width(140.dp),
                shape = RoundedCornerShape(16.dp),
                colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
                elevation = CardDefaults.cardElevation(defaultElevation = 2.dp),
            ) {
                Column(modifier = Modifier.padding(14.dp)) {
                    Text(
                        text = metric.value,
                        style = MaterialTheme.typography.headlineSmall,
                        fontWeight = FontWeight.Bold,
                        color = StaffPrimary,
                    )
                    Spacer(modifier = Modifier.height(4.dp))
                    Text(
                        text = metric.label,
                        style = MaterialTheme.typography.bodySmall,
                        color = StaffOnSurfaceVariant,
                    )
                }
            }
        }
    }
}

@Composable
fun StaffChipRow(
    chips: List<DayChipUi>,
    onChipClick: (String) -> Unit,
    modifier: Modifier = Modifier,
) {
    LazyRow(
        modifier = modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.spacedBy(8.dp),
        contentPadding = PaddingValues(horizontal = 16.dp, vertical = 8.dp),
    ) {
        items(chips) { chip ->
            FilterChip(
                selected = chip.selected,
                onClick = { onChipClick(chip.date) },
                label = {
                    Column(horizontalAlignment = Alignment.CenterHorizontally) {
                        Text(chip.label, style = MaterialTheme.typography.labelMedium)
                        if (chip.count >= 0) {
                            Text(
                                "${chip.count}",
                                style = MaterialTheme.typography.labelSmall,
                            )
                        }
                    }
                },
                colors = FilterChipDefaults.filterChipColors(
                    selectedContainerColor = StaffPrimary,
                    selectedLabelColor = Color.White,
                ),
            )
        }
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun StaffListCard(
    item: ListCardUi,
    onClick: (() -> Unit)? = null,
    modifier: Modifier = Modifier,
) {
    val clickable = onClick != null
    val cardColors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface)
    val cardElevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
    val cardShape = RoundedCornerShape(16.dp)
    val cardBody: @Composable ColumnScope.() -> Unit = {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(16.dp),
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Column(modifier = Modifier.weight(1f)) {
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Text(
                        text = item.title,
                        style = MaterialTheme.typography.titleSmall,
                        fontWeight = FontWeight.SemiBold,
                    )
                    if (item.badge != null) {
                        Spacer(modifier = Modifier.width(8.dp))
                        StaffBadge(text = item.badge, color = item.badgeColor)
                    }
                }
                if (item.subtitle.isNotBlank()) {
                    Spacer(modifier = Modifier.height(4.dp))
                    Text(
                        text = item.subtitle,
                        style = MaterialTheme.typography.bodyMedium,
                        color = MaterialTheme.colorScheme.onSurface,
                    )
                }
                if (item.meta.isNotBlank()) {
                    Spacer(modifier = Modifier.height(4.dp))
                    Text(
                        text = item.meta,
                        style = MaterialTheme.typography.bodySmall,
                        color = StaffOnSurfaceVariant,
                    )
                }
            }
            if (clickable) {
                Icon(
                    imageVector = Icons.AutoMirrored.Filled.KeyboardArrowRight,
                    contentDescription = null,
                    tint = StaffOnSurfaceVariant,
                )
            }
        }
    }
    if (clickable) {
        Card(
            onClick = onClick!!,
            modifier = modifier.fillMaxWidth(),
            shape = cardShape,
            colors = cardColors,
            elevation = cardElevation,
            content = cardBody,
        )
    } else {
        Card(
            modifier = modifier.fillMaxWidth(),
            shape = cardShape,
            colors = cardColors,
            elevation = cardElevation,
            content = cardBody,
        )
    }
}

@Composable
fun StaffBadge(text: String, color: BadgeColor) {
    val bg = when (color) {
        BadgeColor.SUCCESS -> StaffSuccess.copy(alpha = 0.15f)
        BadgeColor.WARNING -> StaffWarning.copy(alpha = 0.2f)
        BadgeColor.ERROR -> StaffError.copy(alpha = 0.15f)
        BadgeColor.PRIMARY -> StaffPrimary.copy(alpha = 0.12f)
        BadgeColor.NEUTRAL -> StaffOnSurfaceVariant.copy(alpha = 0.12f)
    }
    val fg = when (color) {
        BadgeColor.SUCCESS -> StaffSuccess
        BadgeColor.WARNING -> StaffWarning
        BadgeColor.ERROR -> StaffError
        BadgeColor.PRIMARY -> StaffPrimary
        BadgeColor.NEUTRAL -> StaffOnSurfaceVariant
    }
    Text(
        text = text,
        modifier = Modifier
            .clip(RoundedCornerShape(8.dp))
            .background(bg)
            .padding(horizontal = 8.dp, vertical = 2.dp),
        style = MaterialTheme.typography.labelSmall,
        color = fg,
        fontWeight = FontWeight.Medium,
    )
}

@Composable
fun StaffMenuCard(
    title: String,
    items: List<Pair<ImageVector, Pair<String, String>>>,
    onItemClick: (Int) -> Unit,
    modifier: Modifier = Modifier,
) {
    Card(
        modifier = modifier.fillMaxWidth(),
        shape = RoundedCornerShape(20.dp),
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp),
    ) {
        Column {
            Text(
                text = title,
                style = MaterialTheme.typography.titleMedium,
                fontWeight = FontWeight.Bold,
                modifier = Modifier.padding(start = 16.dp, top = 16.dp, end = 16.dp, bottom = 8.dp),
            )
            items.forEachIndexed { index, (icon, texts) ->
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .clickable { onItemClick(index) }
                        .padding(horizontal = 16.dp, vertical = 14.dp),
                    verticalAlignment = Alignment.CenterVertically,
                ) {
                    Box(
                        modifier = Modifier
                            .size(44.dp)
                            .clip(RoundedCornerShape(12.dp))
                            .background(StaffPrimary.copy(alpha = 0.1f)),
                        contentAlignment = Alignment.Center,
                    ) {
                        Icon(icon, contentDescription = null, tint = StaffPrimary)
                    }
                    Spacer(modifier = Modifier.width(14.dp))
                    Column(modifier = Modifier.weight(1f)) {
                        Text(texts.first, style = MaterialTheme.typography.bodyLarge, fontWeight = FontWeight.Medium)
                        if (texts.second.isNotBlank()) {
                            Text(
                                texts.second,
                                style = MaterialTheme.typography.bodySmall,
                                color = StaffOnSurfaceVariant,
                            )
                        }
                    }
                    Icon(
                        Icons.AutoMirrored.Filled.KeyboardArrowRight,
                        contentDescription = null,
                        tint = StaffOnSurfaceVariant,
                    )
                }
                if (index < items.lastIndex) {
                    HorizontalDivider(
                        modifier = Modifier.padding(horizontal = 16.dp),
                        color = MaterialTheme.colorScheme.outlineVariant,
                    )
                }
            }
            Spacer(modifier = Modifier.height(8.dp))
        }
    }
}

@Composable
fun StaffSecondaryButton(
    text: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    enabled: Boolean = true,
) {
    FilledTonalButton(
        onClick = onClick,
        enabled = enabled,
        modifier = modifier.fillMaxWidth(),
        shape = RoundedCornerShape(14.dp),
        colors = ButtonDefaults.filledTonalButtonColors(
            containerColor = StaffPrimary.copy(alpha = 0.12f),
            contentColor = StaffPrimary,
        ),
    ) {
        Text(text)
    }
}

@Composable
fun StaffPrimaryButton(
    text: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    enabled: Boolean = true,
) {
    Button(
        onClick = onClick,
        enabled = enabled,
        modifier = modifier.fillMaxWidth(),
        shape = RoundedCornerShape(14.dp),
        colors = ButtonDefaults.buttonColors(containerColor = StaffPrimary),
        elevation = ButtonDefaults.buttonElevation(defaultElevation = 2.dp),
    ) {
        Text(text)
    }
}

@Composable
fun StaffActionButtons(
    actions: List<ActionUi>,
    onAction: (String) -> Unit,
    modifier: Modifier = Modifier,
) {
    Column(modifier = modifier, verticalArrangement = Arrangement.spacedBy(8.dp)) {
        actions.forEach { action ->
            if (action.id.startsWith("text:")) {
                TextButton(onClick = { onAction(action.id) }) {
                    Text(action.label, color = StaffPrimary)
                }
            } else if (action.id.startsWith("secondary:")) {
                StaffSecondaryButton(text = action.label, onClick = { onAction(action.id) })
            } else {
                StaffPrimaryButton(text = action.label, onClick = { onAction(action.id) })
            }
        }
    }
}

@Composable
fun StaffSearchBar(
    query: String,
    onQueryChange: (String) -> Unit,
    onSearch: () -> Unit,
    modifier: Modifier = Modifier,
) {
    Surface(
        modifier = modifier.fillMaxWidth(),
        color = MaterialTheme.colorScheme.surface,
        shadowElevation = 2.dp,
    ) {
        Row(
            modifier = Modifier.padding(12.dp),
            verticalAlignment = Alignment.CenterVertically,
        ) {
            OutlinedTextField(
                value = query,
                onValueChange = onQueryChange,
                modifier = Modifier.weight(1f),
                placeholder = { Text("Имя, email или телефон") },
                singleLine = true,
                shape = RoundedCornerShape(14.dp),
            )
            Spacer(modifier = Modifier.width(8.dp))
            Button(
                onClick = onSearch,
                shape = RoundedCornerShape(14.dp),
                colors = ButtonDefaults.buttonColors(containerColor = StaffPrimary),
            ) {
                Text("Найти")
            }
        }
    }
}

@Composable
fun StaffLoadingState(message: String = "Загрузка...") {
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .padding(48.dp),
        contentAlignment = Alignment.Center,
    ) {
        Column(horizontalAlignment = Alignment.CenterHorizontally) {
            CircularProgressIndicator(color = StaffPrimary)
            Spacer(modifier = Modifier.height(12.dp))
            Text(message, color = StaffOnSurfaceVariant)
        }
    }
}

@Composable
fun StaffEmptyState(
    message: String,
    icon: ImageVector = Icons.Default.Inbox,
) {
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .padding(40.dp),
        contentAlignment = Alignment.Center,
    ) {
        Column(horizontalAlignment = Alignment.CenterHorizontally) {
            Icon(icon, contentDescription = null, modifier = Modifier.size(56.dp), tint = StaffOnSurfaceVariant)
            Spacer(modifier = Modifier.height(12.dp))
            Text(message, style = MaterialTheme.typography.bodyLarge, color = StaffOnSurfaceVariant)
        }
    }
}

@Composable
fun StaffErrorState(
    message: String,
    onRetry: (() -> Unit)? = null,
    modifier: Modifier = Modifier,
) {
    Card(
        modifier = modifier.fillMaxWidth(),
        shape = RoundedCornerShape(16.dp),
        colors = CardDefaults.cardColors(containerColor = StaffError.copy(alpha = 0.08f)),
    ) {
        Column(modifier = Modifier.padding(16.dp)) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Icon(Icons.Default.ErrorOutline, contentDescription = null, tint = StaffError)
                Spacer(modifier = Modifier.width(8.dp))
                Text(message, color = StaffError, style = MaterialTheme.typography.bodyMedium)
            }
            if (onRetry != null) {
                Spacer(modifier = Modifier.height(8.dp))
                TextButton(onClick = onRetry) { Text("Повторить", color = StaffPrimary) }
            }
        }
    }
}

@Composable
fun StaffInfoBanner(text: String, color: Color = StaffAccentBlue) {
    Card(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(14.dp),
        colors = CardDefaults.cardColors(containerColor = color.copy(alpha = 0.1f)),
    ) {
        Text(
            text = text,
            modifier = Modifier.padding(14.dp),
            style = MaterialTheme.typography.bodyMedium,
            color = color,
        )
    }
}

File: MatskocaAdmin/MatskocaAdmin/Components/StaffComponents.swift
import SwiftUI

struct StaffSectionTitle: View {
    let title: String

    var body: some View {
        Text(title)
            .font(.title3)
            .fontWeight(.bold)
            .padding(.top, 4)
            .padding(.bottom, 8)
            .frame(maxWidth: .infinity, alignment: .leading)
    }
}

struct StaffHeroCard: View {
    let title: String
    let subtitle: String

    var body: some View {
        VStack(alignment: .leading, spacing: 6) {
            Text(title)
                .font(.title2)
                .fontWeight(.bold)
                .foregroundStyle(StaffColors.onPrimary)
            Text(subtitle)
                .font(.body)
                .foregroundStyle(StaffColors.onPrimary.opacity(0.9))
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(20)
        .background(StaffColors.primary)
        .clipShape(RoundedRectangle(cornerRadius: 20))
        .shadow(color: .black.opacity(0.08), radius: 4, y: 2)
    }
}

struct StaffMetricsRow: View {
    let metrics: [MetricUi]

    var body: some View {
        ScrollView(.horizontal, showsIndicators: false) {
            HStack(spacing: 10) {
                ForEach(metrics) { metric in
                    VStack(alignment: .leading, spacing: 4) {
                        Text(metric.value)
                            .font(.title2)
                            .fontWeight(.bold)
                            .foregroundStyle(StaffColors.primary)
                        Text(metric.label)
                            .font(.caption)
                            .foregroundStyle(StaffColors.onSurfaceVariant)
                    }
                    .frame(width: 140, alignment: .leading)
                    .padding(14)
                    .background(StaffColors.surface)
                    .clipShape(RoundedRectangle(cornerRadius: 16))
                    .shadow(color: .black.opacity(0.06), radius: 2, y: 1)
                }
            }
        }
    }
}

struct StaffChipRow: View {
    let chips: [DayChipUi]
    let onChipClick: (String) -> Void

    var body: some View {
        ScrollView(.horizontal, showsIndicators: false) {
            HStack(spacing: 8) {
                ForEach(chips) { chip in
                    Button {
                        onChipClick(chip.date)
                    } label: {
                        VStack(spacing: 2) {
                            Text(chip.label)
                                .font(.subheadline)
                            if chip.count >= 0 {
                                Text("\(chip.count)")
                                    .font(.caption2)
                            }
                        }
                        .padding(.horizontal, 12)
                        .padding(.vertical, 8)
                        .background(chip.selected ? StaffColors.primary : StaffColors.surface)
                        .foregroundStyle(chip.selected ? StaffColors.onPrimary : StaffColors.onBackground)
                        .clipShape(Capsule())
                        .overlay(
                            Capsule().stroke(StaffColors.onSurfaceVariant.opacity(0.3), lineWidth: chip.selected ? 0 : 1)
                        )
                    }
                    .buttonStyle(.plain)
                }
            }
            .padding(.horizontal, 16)
            .padding(.vertical, 8)
        }
    }
}

struct StaffBadge: View {
    let text: String
    let color: BadgeColor

    var body: some View {
        Text(text)
            .font(.caption2)
            .fontWeight(.medium)
            .padding(.horizontal, 8)
            .padding(.vertical, 2)
            .background(backgroundColor)
            .foregroundStyle(foregroundColor)
            .clipShape(RoundedRectangle(cornerRadius: 8))
    }

    private var backgroundColor: Color {
        switch color {
        case .success: return StaffColors.success.opacity(0.15)
        case .warning: return StaffColors.warning.opacity(0.2)
        case .error: return StaffColors.error.opacity(0.15)
        case .primary: return StaffColors.primary.opacity(0.12)
        case .neutral: return StaffColors.onSurfaceVariant.opacity(0.12)
        }
    }

    private var foregroundColor: Color {
        switch color {
        case .success: return StaffColors.success
        case .warning: return StaffColors.warning
        case .error: return StaffColors.error
        case .primary: return StaffColors.primary
        case .neutral: return StaffColors.onSurfaceVariant
        }
    }
}

struct StaffListCard: View {
    let item: ListCardUi
    var onClick: (() -> Void)?

    var body: some View {
        Group {
            if let onClick {
                Button(action: onClick) { cardContent }
                    .buttonStyle(.plain)
            } else {
                cardContent
            }
        }
    }

    private var cardContent: some View {
        HStack(alignment: .center) {
            VStack(alignment: .leading, spacing: 4) {
                HStack(alignment: .center, spacing: 8) {
                    Text(item.title)
                        .font(.subheadline)
                        .fontWeight(.semibold)
                        .foregroundStyle(StaffColors.onBackground)
                    if let badge = item.badge {
                        StaffBadge(text: badge, color: item.badgeColor)
                    }
                }
                if !item.subtitle.isEmpty {
                    Text(item.subtitle)
                        .font(.body)
                        .foregroundStyle(StaffColors.onBackground)
                }
                if !item.meta.isEmpty {
                    Text(item.meta)
                        .font(.caption)
                        .foregroundStyle(StaffColors.onSurfaceVariant)
                }
            }
            Spacer(minLength: 8)
            if onClick != nil {
                Image(systemName: "chevron.right")
                    .foregroundStyle(StaffColors.onSurfaceVariant)
            }
        }
        .padding(16)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(StaffColors.surface)
        .clipShape(RoundedRectangle(cornerRadius: 16))
        .shadow(color: .black.opacity(0.06), radius: 2, y: 1)
    }
}

struct StaffMenuCard: View {
    let title: String
    let items: [(icon: String, title: String, hint: String)]
    let onItemClick: (Int) -> Void

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            Text(title)
                .font(.title3)
                .fontWeight(.bold)
                .padding(.horizontal, 16)
                .padding(.top, 16)
                .padding(.bottom, 8)

            ForEach(Array(items.enumerated()), id: \.offset) { index, item in
                Button {
                    onItemClick(index)
                } label: {
                    HStack(spacing: 14) {
                        Image(systemName: item.icon)
                            .font(.title3)
                            .foregroundStyle(StaffColors.primary)
                            .frame(width: 44, height: 44)
                            .background(StaffColors.primary.opacity(0.1))
                            .clipShape(RoundedRectangle(cornerRadius: 12))

                        VStack(alignment: .leading, spacing: 2) {
                            Text(item.title)
                                .font(.body)
                                .fontWeight(.medium)
                                .foregroundStyle(StaffColors.onBackground)
                            if !item.hint.isEmpty {
                                Text(item.hint)
                                    .font(.caption)
                                    .foregroundStyle(StaffColors.onSurfaceVariant)
                            }
                        }
                        Spacer()
                        Image(systemName: "chevron.right")
                            .foregroundStyle(StaffColors.onSurfaceVariant)
                    }
                    .padding(.horizontal, 16)
                    .padding(.vertical, 14)
                }
                .buttonStyle(.plain)

                if index < items.count - 1 {
                    Divider().padding(.horizontal, 16)
                }
            }
            Spacer().frame(height: 8)
        }
        .background(StaffColors.surface)
        .clipShape(RoundedRectangle(cornerRadius: 20))
        .shadow(color: .black.opacity(0.06), radius: 2, y: 1)
    }
}

struct StaffPrimaryButton: View {
    let text: String
    let action: () -> Void
    var enabled: Bool = true

    var body: some View {
        Button(action: action) {
            Text(text)
                .fontWeight(.medium)
                .frame(maxWidth: .infinity)
                .padding(.vertical, 12)
                .foregroundStyle(StaffColors.onPrimary)
                .background(enabled ? StaffColors.primary : StaffColors.primary.opacity(0.5))
                .clipShape(RoundedRectangle(cornerRadius: 14))
                .contentShape(RoundedRectangle(cornerRadius: 14))
        }
        .buttonStyle(.plain)
        .disabled(!enabled)
    }
}

struct StaffSecondaryButton: View {
    let text: String
    let action: () -> Void
    var enabled: Bool = true

    var body: some View {
        Button(action: action) {
            Text(text)
                .fontWeight(.medium)
                .frame(maxWidth: .infinity)
                .padding(.vertical, 12)
                .foregroundStyle(StaffColors.primary)
                .background(StaffColors.primary.opacity(enabled ? 0.12 : 0.06))
                .clipShape(RoundedRectangle(cornerRadius: 14))
                .contentShape(RoundedRectangle(cornerRadius: 14))
        }
        .buttonStyle(.plain)
        .disabled(!enabled)
    }
}

struct StaffActionButtons: View {
    let actions: [ActionUi]
    let onAction: (String) -> Void

    var body: some View {
        VStack(spacing: 8) {
            ForEach(actions) { action in
                if action.id.hasPrefix("text:") {
                    Button(action.label) { onAction(action.id) }
                        .foregroundStyle(StaffColors.primary)
                } else if action.id.hasPrefix("secondary:") {
                    StaffSecondaryButton(text: action.label) { onAction(action.id) }
                } else {
                    StaffPrimaryButton(text: action.label) { onAction(action.id) }
                }
            }
        }
    }
}

struct StaffSearchBar: View {
    let query: String
    let onQueryChange: (String) -> Void
    let onSearch: () -> Void

    var body: some View {
        HStack(spacing: 8) {
            TextField("Имя, email или телефон", text: Binding(
                get: { query },
                set: onQueryChange
            ))
            .textFieldStyle(.roundedBorder)

            Button("Найти", action: onSearch)
                .padding(.horizontal, 12)
                .padding(.vertical, 8)
                .background(StaffColors.primary)
                .foregroundStyle(StaffColors.onPrimary)
                .clipShape(RoundedRectangle(cornerRadius: 14))
        }
        .padding(12)
        .background(StaffColors.surface)
        .shadow(color: .black.opacity(0.06), radius: 2, y: 1)
    }
}

struct StaffLoadingState: View {
    var message: String = "Загрузка..."

    var body: some View {
        VStack(spacing: 12) {
            ProgressView()
                .tint(StaffColors.primary)
            Text(message)
                .foregroundStyle(StaffColors.onSurfaceVariant)
        }
        .frame(maxWidth: .infinity)
        .padding(48)
    }
}

struct StaffEmptyState: View {
    let message: String
    var icon: String = "tray"

    var body: some View {
        VStack(spacing: 12) {
            Image(systemName: icon)
                .font(.system(size: 56))
                .foregroundStyle(StaffColors.onSurfaceVariant)
            Text(message)
                .font(.body)
                .foregroundStyle(StaffColors.onSurfaceVariant)
        }
        .frame(maxWidth: .infinity)
        .padding(40)
    }
}

struct StaffErrorState: View {
    let message: String
    var onRetry: (() -> Void)?

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            HStack(alignment: .top, spacing: 8) {
                Image(systemName: "exclamationmark.circle")
                    .foregroundStyle(StaffColors.error)
                Text(message)
                    .foregroundStyle(StaffColors.error)
                    .font(.body)
            }
            if let onRetry {
                Button("Повторить", action: onRetry)
                    .foregroundStyle(StaffColors.primary)
            }
        }
        .padding(16)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(StaffColors.error.opacity(0.08))
        .clipShape(RoundedRectangle(cornerRadius: 16))
    }
}

struct StaffInfoBanner: View {
    let text: String
    var color: Color = StaffColors.accentBlue

    var body: some View {
        Text(text)
            .font(.body)
            .foregroundStyle(color)
            .padding(14)
            .frame(maxWidth: .infinity, alignment: .leading)
            .background(color.opacity(0.1))
            .clipShape(RoundedRectangle(cornerRadius: 14))
    }
}
